raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -975,6 +980,18 @@ void assertFetchRequestData(
         assertEquals(fetchOffset, fetchPartition.fetchOffset());
         assertEquals(lastFetchedEpoch, fetchPartition.lastFetchedEpoch());
         assertEquals(localId.orElse(-1), request.replicaId());
+
+        // Assert that voters have flushed up to the fetch offset
+        if (localId.isPresent() && voters.contains(localId.getAsInt())) {
```

Comment:
Is there anything we can assert in the converse case to verify that the log is not flushed?

Reply:
If someone removed the optimization, it might be a notable regression in performance, but no tests would fail. I think that makes it worthwhile.

Reply:
Okay. Added a check for observers.